app/domain/authentication/authn_jwt/identity_providers/identity_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,80 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Class for providing jwt identity from the decoded token from the field specified in a secret
+      class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+        def initialize(
+          authentication_parameters,
+          fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+          resource_class: ::Resource,
+          logger: Rails.logger
+        )
+          @logger = logger
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @authentication_parameters = authentication_parameters
+        end
+
+        def jwt_identity
+          return @jwt_identity if @jwt_identity
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CheckingIdentityFieldExists.new(token_field_name))
+          @jwt_identity = decoded_token[token_field_name]
+          if @jwt_identity.blank?
+            raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken, token_field_name
+          end
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FoundJwtFieldInToken.new(token_field_name, @jwt_identity))
+          @jwt_identity
+        end
+
+        # Checks if variable that defined from which field in decoded token to get the id is configured
+        def identity_available?
+          return @identity_available if defined?(@identity_available)
+
+          @identity_available = identity_field_variable.present?
+        end
+
+        # This method is for the authenticator status check, unlike 'identity_available?' it checks if the
+        # secret value is not empty too
+        def identity_configured_properly?
+          identity_available? && token_field_name.blank?
+        end
+
+        private
+
+        def variable_id
+          @authentication_parameters.authn_jwt_variable_id_prefix
+        end
+
+        def decoded_token
+          @authentication_parameters.decoded_token
+        end
+
+        def identity_field_variable
+          @identity_field_variable ||= @resource_class[token_id_field_variable_id]
+        end
+
+        def token_field_name
+          token_id_field_secret
+        end
+
+        def token_id_field_secret
+          return @token_id_field_secret if @token_id_field_secret
+
+          @token_id_field_secret = conjur_secret(token_id_field_variable_id)
+          @logger.info(LogMessages::Authentication::AuthnJwt::RetrievedResourceValue.new(@token_id_field_secret, token_id_field_variable_id))
+          @token_id_field_secret
+        end
```

Comment:
we don't need 2 methods for this. either remove the `token_field_name` method and use `token_id_field_secret` in the rest of the code or remove the  `token_field_name` method and rename the `token_id_field_secret` method to `token_field_name`

Reply:
Done